client/my-sites/theme/theme-style-variations/index.tsx

```diff
@@ -0,0 +1,118 @@
+import { Button } from '@automattic/components';
+import { PremiumBadge } from '@automattic/design-picker';
+import { useLayoutEffect, useRef, useState } from '@wordpress/element';
+import classNames from 'classnames';
+import { translate } from 'i18n-calypso';
+import AsyncLoad from 'calypso/components/async-load';
+import type { StyleVariation } from '@automattic/design-picker/src/types';
+import type { TranslateResult } from 'i18n-calypso';
+
+import './style.scss';
+
+interface ThemeStyleVariationsProps {
+	description: TranslateResult;
+	selectedVariation: StyleVariation;
+	variations: StyleVariation[];
+	onClick: ( variation: StyleVariation ) => void;
+}
+
+const ThemeStyleVariations = ( {
+	description,
+	selectedVariation,
+	variations,
+	onClick,
+}: ThemeStyleVariationsProps ) => {
+	const observerRef = useRef< HTMLDivElement | null >( null );
+	const [ collapsibleMaxHeight, setCollapsibleMaxHeight ] = useState< number | undefined >();
```

Comment:
[nit] Maybe we don't need to store the max height in the state 🤔

```js
const maxHeight = isCollapsed ? observerRef.current?.firstChild?.offsetHeight : observerRef.current?.scrollHeight;
```

Reply:
Hmmm I think we need to update the max height in the `ResizeObsever` callback as well since the style variation size dimensions might change according to the viewport 😞 We can optimize this further later 🙂

Reply:
Got it. I forgot the style variation size dimensions might also change 😓